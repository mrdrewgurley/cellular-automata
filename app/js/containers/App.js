import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import * as BoardActions from '../actions/BoardActions'
import Board, { BoardShape } from '../components/Board'
import Controls from '../components/Controls'

const App = props => (
  <div>
    <Board
      data={props.board}
      flip={props.actions.flip}
    />
    <Controls
      actions={props.actions}
      controls={props.controls}
    />
  </div>
)

App.propTypes = {
  actions: PropTypes.object.isRequired,
  board: BoardShape.isRequired,
  controls: PropTypes.object.isRequired
}

export default connect(
  state => ({ ...state }),
  dispatch => ({ actions: bindActionCreators(BoardActions, dispatch) })
)(App)
