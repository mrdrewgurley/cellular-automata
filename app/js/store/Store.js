import { createStore } from 'redux'
import reducers from '../reducers'

/**
 * @desc returns the applications store
 * @param {object} intialState - default dataset for the application store
 * @return {object}
 */
export default initialState =>
  createStore(reducers, initialState)
