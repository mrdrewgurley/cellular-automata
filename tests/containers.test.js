import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import App from '../app/js/containers/App'
import Store from '../app/js/store/Store'

const store = Store()
store.getState().board = [[false,true],[true,false]]

describe("App Container", () => {
  it("renders with proper defaults", () => {
    const AppContainer = renderer
      .create(<Provider store={store}>
        <App />
      </Provider>)
      .toJSON()

    expect(AppContainer).toMatchSnapshot()
  })
})
