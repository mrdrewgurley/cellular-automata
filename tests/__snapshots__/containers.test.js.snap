// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container renders with proper defaults 1`] = `
<div>
  <div
    className="board"
  >
    <div
      className="board-row"
    >
      <div
        className="board-block"
        onMouseDown={[Function]}
        style={
          Object {
            "backgroundColor": null,
          }
        }
      />
      <div
        className="board-block"
        onMouseDown={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      />
    </div>
    <div
      className="board-row"
    >
      <div
        className="board-block"
        onMouseDown={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      />
      <div
        className="board-block"
        onMouseDown={[Function]}
        style={
          Object {
            "backgroundColor": null,
          }
        }
      />
    </div>
  </div>
  <div
    className="controls uk-button-group"
  >
    <button
      className="uk-button uk-button-default uk-margin-right"
      onClick={[Function]}
    >
      START
    </button>
    <button
      className="uk-button uk-button-default uk-margin-right"
      onClick={[Function]}
    >
      RESET
    </button>
    <button
      className="uk-button uk-button-default"
      onClick={[Function]}
    >
      CLEAR
    </button>
  </div>
</div>
`;
